### PR TITLE
Specify change password GraphQL API

### PR DESCRIPTION
**Pull request by IniZio**
ref DEV-1642

### CODE REVIEW

File: docs/specs/change-password.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Change Password
```
Comment (louischan-oursky) [inline docs/specs/change-password.md:1]: - Rename this spec to `admin-api-change-password.md`
- Rename the title to `Change Password in Admin API`

File: docs/specs/change-password.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+input ResetPasswordInput {
+  # other fields...
+
+  # Email will be sent to user with new password
+  sendPassword: Boolean
+  # User will be forced to change password on next login
+  forceChangeOnLogin: Boolean
+  # Generate password and send it to user through email
+  generateAndSendPassword: Boolean
```
Comment (louischan-oursky) [inline docs/specs/change-password.md:47]: Can you include the full original ResetPasswordInput and specify it again? It is easier to understand to the full picture.

Do not combine generatePassword with sendPassword.

```graphql
input ResetPasswordInput {
  userID: ID!
  # Optional. It was required previously. If password is not given, then it is generated by Authgear
  password: String
  # Optional. Default is false.
  sendPassword: Boolean
  # Optional. Default is false.
  markPasswordAsExpired: Boolean
}
```
With this design, then we do not need validate on the input, every input has meaning.

```
{
  "userID": "userID"
}
```

This means generate password

```
{
  "userID": "userID",
  "password": "password"
}
```

This means change the password to the given value.

The `sendPassword` and `markPasswordAsExpired` flags are now just optional flags, and composable.

File: docs/specs/change-password.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+```
+
+#### Error Response
+
+|Description|Name|Reason|Info|
+|---|---|---|---|
+|Email identity not found, required if `sendPassword` or `generateAndSendPassword` is set to `true`.|`NotFound`|`EmailIdentityNotFound`|-|
```
Comment (louischan-oursky) [inline docs/specs/change-password.md:55]: Invalid|SendPasswordNoTarget

File: docs/specs/admin-api-change-password.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  # Optional. Default is false. If set to true, then user will be forced to change password on next login.
+  markPasswordAsExpired: Boolean
+}
+```
+
+#### Error Response
+
+|Description|Name|Reason|Info|
+|---|---|---|---|
+|Email identity not found, required if `sendPassword` is set to `true` or password is auto-generated.|`Invalid`|`SendPasswordNoTarget`|-|
```
Comment (louischan-oursky) [inline docs/specs/admin-api-change-password.md:55]: Change the description to 

> If `sendPassword` is true, and the user has no verified email to send the password to

That is, the error will not be thrown if `sendPassword` is false, even if the password is generated.

File: docs/specs/admin-api-change-password.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+### Force Change Password on Next Login
+
+A `markPasswordAsExpired` mutation will be added to the Admin API for forcing user to change their password on next login.
+
+```graphql
+type Mutation {
+  # other root fields...
+
+  markPasswordAsExpired(
```
Comment (louischan-oursky) [inline docs/specs/admin-api-change-password.md:65]: Rename this to `setPasswordExpired`

File: docs/specs/admin-api-change-password.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    input: MarkPasswordAsExpiredInput!
```
Comment (louischan-oursky) [inline docs/specs/admin-api-change-password.md:66]: `SetPasswordExpiredInput`

File: docs/specs/admin-api-change-password.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  ): MarkPasswordAsExpiredPayload!
+}
+
+input MarkPasswordAsExpiredInput {
+  userID: ID!
```
Comment (louischan-oursky) [inline docs/specs/admin-api-change-password.md:71]: Add the field `expired: Boolean!`. The goal is to allow to the developer to revert the change.

Comment (IniZio) [reply docs/specs/admin-api-change-password.md:71]: Updated

**Review by louischan-oursky — APPROVED** [review]
